homeworks/07-mars-viewer/src/components/MarsViewer/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { BaseSyntheticEvent } from "react";
+import { useAppDispatch, useAppSelector } from "../../app/hooks";
+import { marsReducer, fetchPhotosBySol } from "../../app/reducers/mars";
+import { RootState } from "../../app/store";
+import { Photo } from "../Photo";
+import s from "./styles.module.css";
+
+export const MarsViewer = () => {
+  const dispatch = useAppDispatch();
+
+  const onChange = (event: BaseSyntheticEvent) => {
+    const inputValue = parseInt(event.target.value);
+    if (inputValue) {
+      dispatch(marsReducer.actions.changeSelectedSol(inputValue));
+    }
+  };
+
+  const mars = useAppSelector((state: RootState) => state.mars);
+
+  const sol = mars.sols.find((sol) => sol.num === mars.selectedSol);
+
+  const onLoad = () => {
+    dispatch(fetchPhotosBySol(mars.selectedSol));
+  };
+
+  return (
+    <div className={s.viewer}>
+      <p>Select Sol and press "load"!</p>
+      <input
+        className={s.input}
+        onChange={onChange}
+        type="number"
+        value={mars.selectedSol}
+      />
+      <button onClick={onLoad}>load</button>
```

Comment:
In example app http://mars-viewer.surge.sh/ load button being disabled based on request state while it's loading, so user can not make a lot of repeating requests by quickly clicking button